keps/sig-api-machinery/5000-api-linting-crd-schema-tooling/README.md

```diff
@@ -265,20 +266,62 @@ Currently, it is proposed that:
+| uniqueItems | Property | Validates compatibility of changes to the uniqueItems of a property. Net new uniqueItems constraints are flagged | No | Yes | 
+| pattern | Property | Validates compatibility of changes to the pattern constraint of a property. Net new pattern constraints and more restrictive validations are flagged | No | Yes |
+| nullable | Property  | Validates compatibility of changes to the nullable constraint of a property. Making a property no longer nullable is flagged | No | Yes |
+| multipleOf | Property | Validates compatibility of changes to the multipleOf constraint of a property. Changes to the multipleOf constraint are flagged | No | Yes |
+| allOf | Property | Validates compatibility of changes to the allOf constraint of a property. Net new and addition of allOf constraints are flagged | No | Yes |
+| oneOf | Property | Validates compatibility of changes to the oneOf constraint of a property. Net new and removal of oneOf constraints are flagged | No | Yes |
+| anyOf | Property | Validates compatibility of changes to the anyOf constraint of a property. Net new and removal of anyOf constraints are flagged | No | Yes |
+| not | Property | Validates compatibility of changes to the not constraint of a property. Net new not constraints are flagged | No | Yes |
+
+Proposed validations are tracked via https://github.com/everettraven/crd-diff/issues/3.
+
+Validations in https://github.com/openshift/crd-schema-checker  both existing and proposed:
+
+| Name | Scope | Description | Implemented | Carry |
+| ---- | ----- | ----------- | ----------- | ----- |
+| NoBools | Property | Validates that CRD properties are not of type `boolean` | Yes | No, if desired this is better suited for a linter in `kal` |
+| NoFloats | Property | Validates that CRD properties are not of type `number` | Yes | No, if desired this is better suited for a linter in `kal` |
+| NoUints | Property | Validates that CRD properties are not of type `uint` | Yes | No, if desired this is better suited for a linter in `kal` |
+| NoFieldRemoval | CRD | Validates that existing fields are not removed from a CRD | Yes | Yes |
+| NoEnumRemoval | Property | Validates that existing enum values are not removed from a property | Yes | Yes |
+| NoMaps | Property | Validates that CRD properties of type `object` do not have an `additionalProperties` field specified | Yes | No, if desired this is better suited for a linter in `kal` |
+| NoDataTypeChange | Property | Validates that property type is not changed | Yes | Yes |
+| MustHaveStatus | CRD | Validates that the CRD has a status subresource | Yes | No, if desired this is better suited for a linter in `kal` |
+| ListsMustHaveSSATags | Property | Validates that lists have the `x-kubernetes-list-type` tag for server side apply | Yes | No, if desired this is better suited for a linter in `kal` |
```

Comment:
Definitely desired, on the KAL wishlist already

Reply:
Updated the reasoning in https://github.com/kubernetes/enhancements/pull/5102/commits/e69b3eda1028594e6d76b513e3183e12bc8aee91